psi4/driver/procrouting/scf_proc/scf_iterator.py

```diff
@@ -192,9 +192,9 @@ def scf_initialize(self):
     if self.attempt_number_ == 1:
         mints = core.MintsHelper(self.basisset())
         if core.get_global_option('RELATIVISTIC') in ['X2C', 'DKH']:
-            mints.set_rel_basisset(self.get_basisset('BASIS_RELATIVISTIC'))
+            mintshelper = self.mintshelper()
+            mintshelper.set_rel_basisset(self.get_basisset('BASIS_RELATIVISTIC'))
```

Comment:
I think stashing the rel_basis on wfn.basissets() is pretty general/robust. Is it the interaction with mintshelper that's troubling?

Reply:
This is a good point. `MintsHelper` does not store a wavefunction object, so that's why I am storing a copy of the relativistic basis in it. So the alternative is storing a `Wavefunction` object in `MintsHelper`. I am not sure which options is more sound. Any suggestion?

Reply:
I think my original comment was in response to your 

> Passing relativistic_basis is done in an awkward way, by piping it into MintsHelper in the SCF code (py-side). What would be a good way to deal with it that is more general/robust?

and me trying to figure out what problem you were raising. Wfn owning a mintshelper owning a Wfn sounds like the old "which comes first? the molecule or the basisset" problem. Maybe mintshelper can get a basissets dict like Wfn https://github.com/psi4/psi4/blob/master/psi4/src/psi4/libmints/wavefunction.h#L90-L91 that gets initialized from Wfn so that each aux basis doesn't need its own slot.

Reply:
Good suggestion!